src/main/java/appeng/client/render/model/MemoryCardBakedModel.java

```diff
@@ -0,0 +1,232 @@
+	{
+		CubeBuilder builder = new CubeBuilder( this.format );
+
+		builder.setTexture( this.texture );
+
+		for( int x = 0; x < 4; x++ )
+		{
+			for( int y = 0; y < 2; y++ )
+			{
+				final AEColor color = this.colorCode[x + y * 4];
+
+				builder.setColorRGB( color.mediumVariant );
+				builder.addCube( 7 + x, 8 + y, 7.5f, 7 + x + 1, 8 + y + 1, 8.5f );
+			}
+		}
+
+		return builder.getOutput();
+	}
+
+	@Override
+	public boolean isAmbientOcclusion()
+	{
+		return this.baseModel.isAmbientOcclusion();
+	}
+
+	@Override
+	public boolean isGui3d()
+	{
+		return this.baseModel.isGui3d();
+	}
+
+	@Override
+	public boolean isBuiltInRenderer()
+	{
+		return this.baseModel.isBuiltInRenderer();
+	}
+
+	@Override
+	public TextureAtlasSprite getParticleTexture()
+	{
+		return this.baseModel.getParticleTexture();
+	}
+
+	@Override
+	public ItemCameraTransforms getItemCameraTransforms()
+	{
+		return this.baseModel.getItemCameraTransforms();
+	}
+
+	@Override
+	public ItemOverrideList getOverrides()
+	{
+		return new ItemOverrideList( Collections.emptyList() )
+		{
+			@Override
+			public IBakedModel handleItemState( IBakedModel originalModel, ItemStack stack, World world, EntityLivingBase entity )
+			{
+				try
+				{
+					if( stack.getItem() instanceof IMemoryCard )
+					{
+						final IMemoryCard memoryCard = (IMemoryCard) stack.getItem();
+						final AEColor[] colorCode = memoryCard.getColorCode( stack );
+
+						return MemoryCardBakedModel.this.modelCache.get( new CacheKey( colorCode ),
+								() -> new MemoryCardBakedModel( MemoryCardBakedModel.this.format, MemoryCardBakedModel.this.baseModel, MemoryCardBakedModel.this.texture, colorCode, MemoryCardBakedModel.this.modelCache ) );
+					}
+				}
+				catch( ExecutionException e )
+				{
+					AELog.error( e );
+				}
+
+				return MemoryCardBakedModel.this;
+			}
+		};
+
+	}
+
+	@Override
+	public Pair<? extends IBakedModel, Matrix4f> handlePerspective( ItemCameraTransforms.TransformType type )
+	{
+		// Delegate to the base model if possible
+		if( this.baseModel instanceof IBakedModel )
+		{
+			IBakedModel pam = this.baseModel;
+			Pair<? extends IBakedModel, Matrix4f> pair = pam.handlePerspective( type );
+			return Pair.of( this, pair.getValue() );
+		}
+		return Pair.of( this, TRSRTransformation.identity().getMatrix() );
+	}
+
+	private static class CacheKey
+	{
+		private final AEColor[] key;
+
+		public CacheKey( AEColor[] key )
+		{
+			this.key = key;
+		}
+
+		@Override
+		public int hashCode()
+		{
+			final int prime = 31;
+			int result = 1;
+			result = prime * result + Arrays.hashCode( this.key );
+			return result;
+		}
+
+		@Override
+		public boolean equals( Object obj )
+		{
+			if( this == obj )
+			{
+				return true;
+			}
+			if( obj == null )
+			{
+				return false;
+			}
+			if( this.getClass() != obj.getClass() )
+			{
+				return false;
+			}
+			CacheKey other = (CacheKey) obj;
+			if( !Arrays.equals( this.key, other.key ) )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1126)